### PR TITLE
docs: add "Quality Assurance" section

### PR DESCRIPTION
**Pull request by mattdibi**
Added "Quality Assurance" section. This is a simple migration of the old documentation. The "System testing" page was rewritten to reflect more recent versions of Kura

### CODE REVIEW

**Comment by mattdibi** [comment]
@marcellorinaldo ?

File: docs/quality-assurance/unit-testing.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## Build-time testing
+
+Build-time testing is further divided into unit testing and integration testing.
+
+Unit testing is focused on testing separate methods or groups of methods, preferably in a single class. This way it can verify the correct operation of difficult-to-reach corner cases.
+
+Integration testing is a more-high-level testing that tests certain functionality on a group of connected services in an approximation of the real environment. It verifies that the services can successfully register in the environment and connect to other services as well as perform their tasks.
+
+Code coverage of the develop branch can be observed in [Jenkins](https://ci.eclipse.org/kura/job/kura-develop/lastBuild/jacoco/).
```
Comment (marcellorinaldo) [inline docs/quality-assurance/unit-testing.md]: This link does not work, shall we user https://ci.eclipse.org/kura/?

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]

File: docs/quality-assurance/unit-testing.md (hunk 1 above)
Comment (mattdibi) [reply docs/quality-assurance/unit-testing.md]: Sorry, totally missed that.

**Review by marcellorinaldo — APPROVED** [review]
